osquery/extension.go

```diff
@@ -417,30 +420,38 @@ func (e *Extension) writeBufferedLogsForType(typ logger.LogType) error {
 			} else {
 				logs = append(logs, string(v))
 				totalBytes += len(v)
+				logIDs = append(logIDs, k)
 			}
-
-			c.Delete() // Note: This advances the cursor
-			k, v = c.First()
+			k, v = c.Next()
 		}
+		return nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "reading buffered logs")
+	}
 
-		if len(logs) == 0 {
-			// Nothing to send
-			return nil
-		}
+	if len(logs) == 0 {
+		// Nothing to send
+		return nil
+	}
 
-		err := e.writeLogsWithReenroll(context.Background(), typ, logs, true)
-		if err != nil {
-			// Returning an error will cancel the
-			// transaction and the logs will not be
-			// deleted.
-			return errors.Wrap(err, "writing logs")
-		}
+	err = e.writeLogsWithReenroll(context.Background(), typ, logs, true)
+	if err != nil {
+		return errors.Wrap(err, "writing logs")
+	}
 
+	// Delete logs that were successfully sent
+	err = e.db.Update(func(tx *bolt.Tx) error {
```

Comment:
can you document how logs are keyed in boltdb?

Reply:
Added that here: https://github.com/kolide/launcher/pull/220/files#diff-cfe59979dfc77c83ca9ee6b3dabc765cR548